src/PDPVerifier.sol

```diff
@@ -106,6 +113,7 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
     // proofset owner has exclusive permission to add and remove roots and delete the proof set
     mapping(uint256 => address) proofSetOwner;
     mapping(uint256 => address) proofSetProposedOwner;
+    mapping(uint256 => uint256) proofSetLastProvenEpoch;
```

Comment:
One thing this variable is missing here is initialization.  As it is the very first call of provePossession will take the interval since last proving to be the current filecoin block number - 0, i.e. 4 years which will cause the fee to max out.  Two approaches both seem fine to me
1. Initialize the variable to current block time upon first add.  Then I will change it to do this initialization upon first call to nextProvingPeriod in #86 
2. When proofSetLastProvenEpoch is unset we say there has been no time since last proof and the fee is 0 for the first call.

Maybe consult with @irenegia to see if there is any preference from fee design standpoint.

In either case you should have a constant NO_PROEN_EPOCH = 0.  When all leaves are removed from the proofset we should set proofSetLastProvenEpoch = NO_PROVEN_EPOCH.  This will handle the case where a proof set starts up deletes everything and starts up again.

Reply:
@ZenGround0 

I have implemented the following changes

1) When an new proof set is created -> `proofSetLastProvenEpoch ` is set to ` NO_PROVEN_EPOCH` i.e. 0 (this is the default I know but just making it explicit)

2) When a proof set is deleted -> `proofSetLastProvenEpoch` is set to `NO_PROVEN_EPOCH`

3)When an existing proofset becomes empty after all it's roots are removed -> `proofSetLastProvenEpoch` is set to `NO_PROVEN_EPOCH`

4) When the first root is added for a `proofset` -> set `proofSetLastProvenEpoch` to current block number as this option sounds the most reasonable to me.

Reply:
Thanks that sounds perfect, I'll take another look

Reply:
One final request here is to add a test ensuring that this variable is set correctly when the last root of a proofset is removed

Reply:
@ZenGround0 Added a test for this. See `testProofSetLastProvenEpochOnRootRemoval`